tests/infrastructure/test_data_manager_stack.py

```diff
@@ -1022,57 +805,60 @@ def test_indexer_lambda_permission_resource_properties(template):
                 "Fn::GetAtt": [Match.string_like_regexp("IndexerLambda*"), "Arn"]
             },
             "Principal": "s3.amazonaws.com",
-            "SourceAccount": {"Ref": "AWS::AccountId"},
+            "SourceAccount": Match.string_like_regexp("\\d{10}"),
```

Comment:
If you pass in `test_func(template, account)` you should be able to get the testing account ID to use for the matches as well if that is more straightforward for the comparison.
```suggestion
            "SourceAccount": account,
```